### PR TITLE
RavenDB-16709: Exception on CSV import

### PR DESCRIPTION
**Pull request by ArieSLV**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-16709/Exception-on-csv-import

### Additional description

We get an exception when importing a CSV file with empty column names.

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- Tests have been added that prove the fix is effective or that the feature works

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Server/Smuggler/Documents/CsvStreamSource.cs [hunk 1]
```diff
@@ -124,7 +124,8 @@ private bool ProcessFieldsIfNeeded()
             {
                 var property = _csvReader.Context.Reader.HeaderRecord[i];
 
-                if (_csvReader.Context.Reader.HeaderRecord[i][0] == '@')
+                if (string.IsNullOrEmpty(_csvReader.Context.Reader.HeaderRecord[i]) == false &&
```
Comment (grisha-kotler) [inline src/Raven.Server/Smuggler/Documents/CsvStreamSource.cs]: Can we use a common function here?

**Review by grisha-kotler — APPROVED** [review]

File: src/Raven.Server/Smuggler/Documents/CsvStreamSource.cs (hunk 1 above)
Comment (ArieSLV) [reply src/Raven.Server/Smuggler/Documents/CsvStreamSource.cs]: Done.

**Review by grisha-kotler — APPROVED** [review]

File: src/Raven.Server/Smuggler/Documents/CsvStreamSource.cs [hunk 2]
```diff
@@ -124,7 +124,7 @@ private bool ProcessFieldsIfNeeded()
             {
                 var property = _csvReader.Context.Reader.HeaderRecord[i];
 
-                if (_csvReader.Context.Reader.HeaderRecord[i][0] == '@')
+                if (_csvReader.Context.Reader.HeaderRecord[i].StartsWith('@'))
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/CsvStreamSource.cs:127]: `_csvReader.Context.Reader.HeaderRecord[i]` cannot be null ?

Comment (ArieSLV) [reply src/Raven.Server/Smuggler/Documents/CsvStreamSource.cs:127]: If CSV has a header, then all `HeaderRecords` entries will be not-null. 
If we have nothing between separator chars, we'll get `string.Empty` value.

**Pull request merged**